openhtf/__init__.py

```diff
@@ -103,8 +103,13 @@ def AddOutputCallbacks(self, *callbacks):
     """Add the given function as an output module to this test."""
     self._test_options.output_callbacks.extend(callbacks)
 
-  def OutputTestRecord(self, record):
+  def OutputTestRecord(self):
     """Feed the record of this test to all output modules."""
+    test_state = self._executor.GetState()
+    if not test_state:
+      return
+
+    record = test_state.GetFinishedRecord()
     for output_cb in self._test_options.output_callbacks:
       output_cb(record)
```

Comment:
while you're here, can you add a `try:` block around each callback so exceptions in output callbacks don't cause this thread to die?